msys tests are failing with segfault

```
Run echo 'NAMES'
NAMES
msys-python3.10.dll
libpython3.10.so
PATHS
D:\a\_temp\4ba34db5-f138-4222-9a5c-89ab68c71b7e: line 4:   819 Segmentation fault      (core dumped) find_libpython -v --candidate-paths
Error: Process completed with exit code 139.
```

[Ci log](https://github.com/ktbarrett/find_libpython/files/9710371/15.txt)

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -180,7 +180,7 @@ jobs:
         msystem:
           # MSYS tests are failing, disable them.
           # See also https://github.com/ktbarrett/find_libpython/issues/39
-          #- MSYS
+          - MSYS
           - MINGW64
     steps:
       - name: Install msys2 (MINGW64)

--- a/noxfile.py
+++ b/noxfile.py
@@ -17,10 +17,34 @@ def tests(session):
 
     # print info
     session.run(
-        "python", "-m", "coverage", "run", "-m", "find_libpython", "--candidate-names"
+        "python",
+        "-m",
+        "coverage",
+        "run",
+        "-m",
+        "find_libpython",
+        "-v",
+        "--platform-info",
     )
     session.run(
-        "python", "-m", "coverage", "run", "-m", "find_libpython", "--candidate-paths"
+        "python",
+        "-m",
+        "coverage",
+        "run",
+        "-m",
+        "find_libpython",
+        "-v",
+        "--candidate-names",
+    )
+    session.run(
+        "python",
+        "-m",
+        "coverage",
+        "run",
+        "-m",
+        "find_libpython",
+        "-v",
+        "--candidate-paths",
     )
     session.run("python", "-m", "coverage", "run", "-m", "find_libpython", "-v")
 

--- a/src/find_libpython/__init__.py
+++ b/src/find_libpython/__init__.py
@@ -38,9 +38,9 @@ from find_libpython._version import __version__  # noqa: F401
 _logger = _getLogger("find_libpython")
 
 _is_apple = sys.platform == "darwin"
-_is_msys = sys.platform == "msys"
+_is_cygwin = sys.platform in ("msys", "cygwin")
 _is_mingw = sys.platform == "mingw"
-_is_windows = os.name == "nt" and not _is_mingw and not _is_msys
+_is_windows = os.name == "nt" and not _is_mingw and not _is_cygwin
 _is_posix = os.name == "posix"
 
 _SHLIB_SUFFIX = _get_config_var("_SHLIB_SUFFIX")
@@ -167,7 +167,7 @@ def candidate_names(suffix=_SHLIB_SUFFIX):
 
     if _is_mingw:
         dlprefix = "lib"
-    elif _is_windows or _is_msys:
+    elif _is_windows or _is_cygwin:
         dlprefix = ""
     else:
         dlprefix = "lib"
@@ -245,7 +245,7 @@ def candidate_paths(suffix=_SHLIB_SUFFIX):
     #
     # But we try other places just in case.
 
-    if _is_windows or _is_msys or _is_mingw:
+    if _is_windows or _is_cygwin or _is_mingw:
         lib_dirs.append(os.path.join(os.path.dirname(sys.executable)))
     else:
         lib_dirs.append(
@@ -260,7 +260,7 @@ def candidate_paths(suffix=_SHLIB_SUFFIX):
 
     lib_basenames = list(candidate_names(suffix=suffix))
 
-    if _is_posix and not _is_msys:
+    if _is_posix and not _is_cygwin:
         for basename in lib_basenames:
             try:
                 libpython = ctypes.CDLL(basename)
@@ -345,7 +345,7 @@ def _finding_libpython():
     _logger.debug("_is_windows = %s", _is_windows)
     _logger.debug("_is_apple = %s", _is_apple)
     _logger.debug("_is_mingw = %s", _is_mingw)
-    _logger.debug("_is_msys = %s", _is_msys)
+    _logger.debug("_is_msys = %s", _is_cygwin)
     _logger.debug("_is_posix = %s", _is_posix)
     for path in candidate_paths():
         _logger.debug("Candidate: %s", path)

--- a/src/find_libpython/__init__.py
+++ b/src/find_libpython/__init__.py
@@ -342,11 +342,6 @@ def _finding_libpython():
     path : str
         Existing path to a libpython.
     """
-    _logger.debug("_is_windows = %s", _is_windows)
-    _logger.debug("_is_apple = %s", _is_apple)
-    _logger.debug("_is_mingw = %s", _is_mingw)
-    _logger.debug("_is_msys = %s", _is_cygwin)
-    _logger.debug("_is_posix = %s", _is_posix)
     for path in candidate_paths():
         _logger.debug("Candidate: %s", path)
         normalized = _normalize_path(path)
@@ -391,11 +386,21 @@ def _cli_find_libpython(cli_op, verbose):
         _print_all(candidate_names())
     elif cli_op == "candidate-paths":
         _print_all(p for p in candidate_paths() if p and os.path.isabs(p))
+    elif cli_op == "platform-info":
+        _log_platform_info()
     else:
         path = find_libpython()
         if path is None:
             return 1
         print(path, end="")
+
+
+def _log_platform_info():
+    _logger.debug("_is_windows = %s", _is_windows)
+    _logger.debug("_is_apple = %s", _is_apple)
+    _logger.debug("_is_mingw = %s", _is_mingw)
+    _logger.debug("_is_msys = %s", _is_cygwin)
+    _logger.debug("_is_posix = %s", _is_posix)
 
 
 def main(args=None):
@@ -431,6 +436,13 @@ def main(args=None):
         dest="cli_op",
         const="candidate-paths",
         help="Print list of candidate paths of libpython.",
+    )
+    group.add_argument(
+        "--platform-info",
+        action="store_const",
+        dest="cli_op",
+        const="platform-info",
+        help="Print information about the platform and exit.",
     )
 
     ns = parser.parse_args(args)

